src/utils/FilterSortPeople.ts

```diff
@@ -0,0 +1,68 @@
+import { Person } from '../types';
+import { QueryParamsType } from '../types/QueryParamsType';
+import { SortFields } from '../types/SortFields';
+
+export const FilterSortPeople = (
+  people: Person[],
+  queryParams: URLSearchParams,
+): Person[] => {
+  const query = queryParams.get(QueryParamsType.Query) || '';
+  const sexFilter = queryParams.get(QueryParamsType.Sex);
+  const centuriesFilter = queryParams.getAll(QueryParamsType.Centuries);
+  const sortField = queryParams.get(QueryParamsType.Sort);
+  const order = queryParams.get(QueryParamsType.Order);
+
+  let filteredPeople = [...people];
```

Comment:
The `filter` method returns a copy of the array, so you don't need to do it manually.

Reply:
> The `filter` method returns a copy of the array, so you don't need to do it manually.

That would break all my logic. I don't see how to do it without this copy.

Reply:
> > The `filter` method returns a copy of the array, so you don't need to do it manually.
> 
> That would break all my logic. I don't see how to do it without this copy.

I suggest you break your function into a few smaller ones. One function is for filtering, second is for sorting.
And pass prepared parameters, your helpers shouldn't know anything about `queryParams`, so you can use in case query, filters, and etc you get from another thing